llvm/lib/CodeGen/GlobalISel/CombinerHelperCompares.cpp

```diff
@@ -74,6 +74,9 @@ bool CombinerHelper::constantFoldFCmp(const GFCmp &FCmp,
   APFloat LHS = LHSCst.getScalarValue();
   APFloat RHS = RHSCst.getScalarValue();
 
+  if (&LHS.getSemantics() != &RHS.getSemantics())
+    return false;
```

Comment:
I think this should have been a malformed query. In the testcase, whatever code looked through the chain to find the constant should have interpreted the 64-bit value, with a bitwise truncate and reinterpret in the new semantics

Reply:
I agree, that should be the final solution to solve this issue. This PR makes sure the semantics are the same before performing the compare with constants and avoids the compiler crash.